tstate/tstate_view.go

```diff
@@ -49,7 +50,11 @@ type TStateView struct {
 	writes      map[string]uint16
 }
 
-func (ts *TState) NewView(scope set.Set[string], storage map[string][]byte) *TStateView {
+func (ts *TState) NewView(scopeKeys set.Set[state.Key], storage map[string][]byte) *TStateView {
```

Comment:
The approach of using `state.Key` as the `key` in `set.Set` means you can now have duplicate (same `Name`) state keys in the same `set.Set`. I recommend using a map for this instead of defining a new struct.